pageserver/src/layered_repository/timeline.rs

```diff
@@ -1448,7 +1498,15 @@ impl Timeline {
         Ok(new_delta_path)
     }
 
-    pub fn compact(&self) -> Result<()> {
+    pub fn compact(&self) -> anyhow::Result<()> {
+        let last_record_lsn = self.get_last_record_lsn();
+
+        // Last record Lsn could be zero in case the timelie was just created
+        if !last_record_lsn.is_valid() {
```

Comment:
I'm not sure how this thing was supposed to work before, presumably, if failed on timeline load for newely created branches and that failure was gracefully handled.

Since now we calculate the timeline size lazily and don't need to call timeline load every time, we stop failing eagerly and get down to compaction to fail.